src/lib/sub-bots/sub-bots.js

```diff
@@ -45,26 +46,25 @@ module.exports = {
     passBot: function() {
         crashHandler.logEvent(TAG, 'passBot');
         return new Promise(function(resolve, reject) {
-            // Probably best not to hit this, should be checked by what's calling it for better handling
-            if (!config.getConfig().subBots) {
-                logger.info(TAG, 'SubBots feature called, but disabled or no subBots on file, rejecting');
-                reject('The sub bot feature is disabled or there are no subBots on file');
-                return;
-            }
 
-            if (currentBots >= config.getConfig().subBotLimit) {
+            if (currentBots >= config.get('subBotLimit')) {
                 logger.info(TAG, 'SubBot requested but maximum number logged on');
                 reject('The maximum number of subBots are currently running');
                 return;
-            } else {
-                currentBots++;
+            }
+
+            // Probably best not to hit this, should be checked by what's calling it for better handling
+            if (!Object.keys(config.get('subBots')).length) {
```

Comment:
(typeof config.get('subBots') !== "object" || !(Object.keys(config.get('subBots')).length > 0))

Reply:
We don't have to test if subBots is an object, because it will always will be(if for some reason it isn't it should just throw a generic error). Second part I don't know if that is necessary, length will always be positive and the only value that we want it to return true is 0.

Reply:
I would be care-full with that. I agree that we want to catch errors and we should probably implement some system where we define different error types which are caught and resolved. However when something is wrong it should throw an error, for instance I found that the environment config key was misspelled and no-one would have noticed. The new system threw an error and it came to my attention.

Reply:
src/lib/discord/discordbot.js line 78. `if (config.getConfig().enviroment === 'development') {`

Reply:
We have a system in logger.js, we throw errors when we need to but it's better if we're the ones throwing them so we can add additional info.

I agree it's not about sweeping these things under the rug though, it's properly handled with reject('The sub bot feature is disabled or there are no subBots on file'); Which is literally what this if statement is testing, it both logs it and very noticeably informs the user.

Handling errors gracefully will often allow easy testing of failure cases too where throwing errors makes testing more complex

Reply:
Oooooh waaaaait I get it now, !0 is true. Yeah ```(!Object.keys(config.get('subBots')).length) ``` is fine, I would like to properly gaurd property calls so an acceptable compromise to me seeing at it is a local var would be to ```(!Object.keys(config.get('subBots')).length) ``` in pass bot and ready functions but before it runs for the first time in ready do ``` if (typeof config.get('subBots') !== "object") { logger.error(TAG, "subBots property not object") } ```

We're still throwing an error, but it's a more graceful exit than a crash as the bot's actually informing us and throwing additional info

Reply:
How far do we want to go with this? It will still throw an generic error when it is passed incorrectly using an env var. Also this is not the only case channels.positions.text/voice, recreationalGames, and communityGames should also be objects. I don’t really see a point to cover all possible senarios of incorrect usage, as it makes things just more complicated.

Reply:
I personally want to go far enough to cover most common instances of errors, as I would much rather throw stack trace alerts on discord and fail gracefully than risk impacting database calls or having to ssh because the bot is constantly restarting and we don't know why. But given that this is an env variable and a less likely point of failure I'mma just gunna merge this rather than argue any more so we can get on with both this project and our lives :)